subprojects/jvm-services/src/test/groovy/org/gradle/internal/jvm/inspection/DefaultJvmMetadataDetectorTest.groovy

```diff
@@ -414,12 +443,16 @@ class DefaultJvmMetadataDetectorTest extends Specification {
          'os.arch': "x86_64",
          'java.vm.name': "Eclipse OpenJ9 VM",
          'java.vm.version': "openj9-0.27.0",
+         'java.vm.vendor': "International Business Machines Corporation",
          'java.runtime.name': "IBM Semeru Runtime Open Edition",
          'java.runtime.version': "${version}-b08"
         ]
     }
 
-    def createExecHandleFactory(Map<String, String> actualProperties) {
+    def createExecHandleFactory(Map<String, String> actualProperties, Boolean validatePropertyKeys = true) {
+        def probedSystemProperties = ProbedSystemProperty.values().findAll { it != ProbedSystemProperty.Z_ERROR }
+        assert !validatePropertyKeys || actualProperties.keySet() == probedSystemProperties.collect { it.systemPropertyKey }.toSet()
```

Comment:
Instead of using a `validate` flag here, I think it would make sense to only validate if the Map is non-empty.